src/controllers/dynakube/oneagent/daemonset/arguments.go

```diff
@@ -45,12 +45,12 @@ func (dsInfo *builderInfo) appendNetworkZoneArg(args []string) []string {
 }
 
 func (dsInfo *builderInfo) appendProxyArg(args []string) []string {
-	if dsInfo.dynakube != nil && dsInfo.dynakube.NeedsOneAgentProxy() {
+	if dsInfo.hasProxy() {
 		return append(args, "--set-proxy=$(https_proxy)")
 	}
 	return args
 }
 
 func (dsInfo *builderInfo) hasProxy() bool {
-	return dsInfo.dynakube != nil && dsInfo.dynakube.Spec.Proxy != nil && (dsInfo.dynakube.Spec.Proxy.ValueFrom != "" || dsInfo.dynakube.Spec.Proxy.Value != "")
+	return dsInfo.dynakube != nil && dsInfo.dynakube.NeedsOneAgentProxy()
```

Comment:
can we remove that hasProxy() function and just use the two expressions directly?

Reply:
IMO its cleaner this way, because the null check makes it easier to test
(also its used multiple times)

Reply:
the `NeedsOneAgentProxy` wraps the `HasProxy` with an extra check for a feature-flag that can turn off proxy just for the oneagent